### DISCUSSION TITLE
Q: 关于昵称 || Q: About nickname

### DISCUSSION THREAD

**Discussion by felixesintot** [walinejs/waline#2099] [Q&A] [answered]
### Describe the bug | 问题描述

我准备在中间件中查询**昵称一样**但是**邮箱不一样**的来实现用户名去重，这有办法做到吗？

### Website URL | 问题网站

https://waline.felixesintot.top

### Where your waline deploy? | 服务部署在哪里？

Vercel (Default)

### Where your comment data store? | 数据存储在哪里？

LeanCloud(https://leancloud.app)

---
### Describe the bug | Problem description

I plan to query in the middleware those whose **nicknames** are the same but **the email addresses are different** to achieve deduplication of user names. Is there a way to do this?

### Website URL | Problem website

https://waline.felixesintot.top

### Where your waline deploy? | Where is the service deployed?

Vercel (Default)

### Where your comment data store? | Where is the data stored?

LeanCloud(https://leancloud.app)

**Comment by lizheming** [comment, marked as answer]
是可以做的，根据查询条件 `.where({nick: 'aaa', mail: ['!=', 'bbb']})` 查询出来是否有用户即可做对应的逻辑。

---
It can be done. According to the query conditions `.where({nick: 'aaa', mail: ['!=', 'bbb']})`, you can query whether there are users and then do the corresponding logic.

**Reply by felixesintot** [reply]
但我发现  /api/user  post上来的信息只有用户名，密码和标签，我应该如何查询到 mail 呢

---
But I found that the information in the /api/user post only has the user name, password and label. How should I query the mail?

**Reply by lizheming** [reply]
https://github.com/walinejs/waline/blob/main/packages/admin/src/pages/register/index.jsx#L54 发布评论和注册的时候都会有 mail

**Reply by felixesintot** [reply]
但我还想获取 update 的时候的 mail 以免用户在注册后把用户名改成别人的

---
But I also want to get the mail during update to prevent users from changing their username to someone else’s

**Reply by lizheming** [reply]
update 的时候 `ctx.state.userInfo` 中有当前用户的信息，包括邮箱

---
When updating, `ctx.state.userInfo` contains the current user’s information, including email
